dotnet-sdk: cross-platform publishing from macOS is broken

### Nixpkgs version

- Unstable (25.05)

### Describe the bug

Attempt to publish any .NET application for a different platform (by using `-r` argument) from macOS results in a error.

### Steps to reproduce

1. `mkdir TestProject`
2. `cd TestProject`
3. `dotnet new console`
4. `dotnet publish -r win-x64`

### Expected behaviour

Publish succeeds.

### Screenshots

_No response_

### Relevant log output

```console
/nix/store/8pb9y18vws4bmbgdli8k4kjx3gsk4l1d-sign-apphost.proj(3,5): error MSB3073: The command "/nix/store/6kcw4mldhi2f36nqgq4var52g0fmwi06-sigtool-0.1.3/bin/codesign -f -s - "[redacted]/obj/Release/net9.0/win-x64/apphost.exe" 2>&1" exited with code 134.
```

### Additional context

_No response_

### System metadata

 - system: `"aarch64-darwin"`
 - host os: `Darwin 24.3.0, macOS 15.3.2`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.24.12`
 - channels(root): `"nixpkgs"`
 - nixpkgs: `/nix/store/9pczav9p5fbrd6adnnx6zrl73jbn6a6j-source`

### Are you using nix-darwin?

Yes, I am using nix-darwin.

### Notify maintainers

@kuznero @mdarocha @corngood 

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+bug%22+label%3A%226.topic%3A+darwin%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/development/compilers/dotnet/sign-apphost.proj
+++ b/pkgs/development/compilers/dotnet/sign-apphost.proj
@@ -1,11 +1,17 @@
 <Project>
-  <Target Name="SignAppHost" AfterTargets="_CreateAppHost" Condition="'$(AppHostIntermediatePath)' != ''">
+  <Target Name="SignAppHost" AfterTargets="_CreateAppHost"
+          Condition="'$(AppHostIntermediatePath)' != ''
+                     and ($(RuntimeIdentifier.StartsWith('osx')) or $(AppHostRuntimeIdentifier.StartsWith('osx')))">
     <Exec Command='@codesign@ -f -s - "$(AppHostIntermediatePath)" 2>&amp;1'/>
   </Target>
-  <Target Name="UnsignBundle" BeforeTargets="GenerateSingleFileBundle" Condition="'$(PublishedSingleFileName)' != ''">
+  <Target Name="UnsignBundle" BeforeTargets="GenerateSingleFileBundle"
+          Condition="'$(PublishedSingleFileName)' != ''
+                     and ($(RuntimeIdentifier.StartsWith('osx')) or $(AppHostRuntimeIdentifier.StartsWith('osx')))">
     <Exec Command='@codesign@ --remove-signature "@(FilesToBundle)" 2>&amp;1' Condition="'%(FilesToBundle.RelativePath)' == '$(PublishedSingleFileName)'"/>
   </Target>
-  <Target Name="SignBundle" AfterTargets="GenerateSingleFileBundle" Condition="'$(PublishedSingleFilePath)' != ''">
+  <Target Name="SignBundle" AfterTargets="GenerateSingleFileBundle"
+          Condition="'$(PublishedSingleFilePath)' != ''
+                     and ($(RuntimeIdentifier.StartsWith('osx')) or $(AppHostRuntimeIdentifier.StartsWith('osx')))">
     <Exec Command='@codesign@ -f -s - "$(PublishedSingleFilePath)" 2>&amp;1'/>
   </Target>
 </Project>

